script.R

```diff
@@ -0,0 +1,136 @@
+  if (length(refs) == 0) {
+    cli::cli_inform("Empty `.revdeprefs.yaml` file")
+    cli::cli_inform("This indicates all reverse dependencies from CRAN.")
+  }
+
+  if (!is.character(refs) && !is.null(refs)) {
+    cli::cli_abort("Unknown structure of `.revdeprefs.yaml` file. Returning.")
+    return(NULL)
+  }
+}
+cli::cli_inform("References used:")
+cli::cli_bullets(refs)
+
+# init
+cli::cli_h1("Initiating...")
+
+## revdepcheck
+cli::cli_h2("revdepcheck...")
+revdepcheck::revdep_reset()
+unlink("./revdep/", recursive = TRUE)
+revdepcheck:::db_disconnect(".")
+usethis::use_revdep()
+revdepcheck:::db_setup(".")
+
+## clean & refresh cache
+cli::cli_h2("crancache...")
+crancache::crancache_clean()
+invisible(crancache::available_packages(repos = c("CRAN" = "https://cloud.r-project.org")))
+
+crancache_dir <- crancache::get_cache_package_dirs()[["other/source"]]
+
+# include refs in revdepcheck
+cli::cli_h1("Adding refs to revdepcheck...")
+## for ref in refs:
+## - download package sources
+## - build .tar.gz file
+## - move .tar.gz file to cache dir and refresh cashe
+## - add to revdep
+for (ref in refs) {
+  cli::cli_inform(sprintf("Adding %s ...", ref))
+
+  ref_parsed <- pkgdepends::parse_pkg_ref(ref)
+  pkg <- ref_parsed$package
+
+  if (!is(ref_parsed, "remote_ref_standard") && !is(ref_parsed, "remote_ref_cran")) {
+    pkgcache::pkg_cache_delete_files(package = pkg)
+
+    temp_download_dest_dir <- tempfile()
+    dir.create(temp_download_dest_dir)
+    x <- pak::pkg_download(ref, dest_dir = temp_download_dest_dir)
+
+    if (file.exists(x$fulltarget)) {
+      targz_path <- x$fulltarget
+    } else if (file.exists(x$fulltarget_tree)) {
+      targz_path <- tempfile(fileext = ".tar.gz")
+      if (file.info(x$fulltarget_tree)$isdir) {
+        pkgbuild::build(
+          file.path(x$fulltarget_tree, "package"),
+          dest_path = targz_path,
+          binary = FALSE,
+          vignettes = FALSE
+        )
+      } else {
+        untarred_dir <- tempfile()
+        dir.create(untarred_dir)
+        untar(x$fulltarget_tree, exdir = untarred_dir)
+        sources_dir <- list.dirs(untarred_dir, recursive = FALSE)[1]
+        pkgbuild::build(sources_dir, dest_path = targz_path, binary = FALSE, vignettes = FALSE)
+        unlink(untarred_dir, recursive = TRUE)
+      }
+    }
+
+    # copy .tar.gz file into pkgcache dir
+    target_path <- file.path(crancache_dir, basename(x$fulltarget))
+    file.copy(targz_path, target_path)
+    unlink(targz_path)
+
+    cranlike::update_PACKAGES(crancache_dir)
+    cli::cli_inform("Added to crancache!")
+  }
+
+  revdepcheck::revdep_add(packages = pkg)
+  cli::cli_inform("Added to revdep todo!")
+}
+
+cli::cli_inform("revdep todo")
+print(revdepcheck::revdep_todo())
+
+# add cache to repos
+options("repos" = c(crancache:::get_crancache_repos(), "CRAN" = "https://cloud.r-project.org"))
+
+cli::cli_h1("Executing revdepcheck...")
+revdepcheck::revdep_check(num_workers = 4)
```

Comment:
do you happen to know how many workers can we use?

Reply:
maybe it's worth to parametrise this in the action definition

Reply:
I think the runner will have 4 CPUs according to [this](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories). I'll double check.

Reply:
Yes, it's 4 CPUs, added a relevant parameter. Now there's [this error](https://github.com/insightsengineering/tern/actions/runs/8723625698/job/23932268263).

Reply:
That error has been resolved